pages/data-providers/data-provider.jsx

```diff
@@ -1,7 +1,133 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
 
 import DataProviderPageTemplate from 'templates/data-provider'
+import useDebouncedEffect from 'hooks/use-debounced-effect'
 
-const DataProviderPage = () => <DataProviderPageTemplate />
+let timeout = null
+
+// TODO: Pages should be just a thin layer of NextJS.
+//       Find a better place for it.
+const useDiscoveryService = (defaultDataProviderUrl) => {
+  const [dataProviderUrl, setDataProviderUrl] = useState(defaultDataProviderUrl)
+  const [discoveredUrls, setDiscoveredUrls] = useState(null)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleDataProviderUrlChange = (newUrl) => {
+    setDiscoveredUrls(null)
```

Comment:
This is the wrong place for the line, state resetting should be debounced.

The desired user flow is the following:
1. the user starts typing
   - we immediately update the string (because the input is controlled)
2. the user stops typing
   - we send the request

To track the difference between 1st and 2nd, we set a timeout after each change that sends the request. If the user types before the timeout callback called, the timeout clears and starts again.

For the message itself:

1. When the request is sent, the message should reset to the default (or to 'loading', should be considered)
2. When the response is retrieved, the correct message is

Reply:
The reason for this line was to remove flickering message. setDiscoveredUrls() is called after setIsLoading() and it resulted to displaying wrong url message for short amount of time. 

This can be avoided if I use `useReducer(store)` as you suggested in different comment.